server/src/main/java/org/elasticsearch/cluster/routing/ShardMovementWriteLoadSimulator.java

```diff
@@ -87,9 +92,46 @@ public Map<String, NodeUsageStatsForThreadPools> simulatedNodeUsageStatsForThrea
                 adjustedNodeUsageStatsForThreadPools.put(entry.getKey(), entry.getValue());
             }
         }
+
+        // Add `NodeUsageStatsForThreadPools` for any nodes not present in the original `NodeUsageStatsForThreadPools` map.
+        addUsageStatsForAnyNodesNotPresentInOriginalNodeUsageStatsForThreadPools(adjustedNodeUsageStatsForThreadPools);
+
         return Collections.unmodifiableMap(adjustedNodeUsageStatsForThreadPools);
     }
 
+    private void addUsageStatsForAnyNodesNotPresentInOriginalNodeUsageStatsForThreadPools(
+        Map<String, NodeUsageStatsForThreadPools> nodeUsageStatsForThreadPools
+    ) {
+        // Assume the new node has the same size thread pool as the largest existing node
+        final OptionalInt largestWriteThreadPool = originalNodeUsageStatsForThreadPools.values()
```

Comment:
Right now we have to build the new node basis every time we calculated an updated ClusterInfo. Could we instead immediately add a new 0 utilization node to `originalNodeUsageStatsForThreadPools`? Then we can move forward always expecting an entry in `originalNodeUsageStatsForThreadPools` for every node.